R/odbc-connection.R

```diff
@@ -14,11 +14,10 @@ OdbcConnection <- function(
     timeout = Inf,
     dbms.name = NULL,
     attributes = NULL,
-    .connection_string = NULL
+    .connection_string = NULL,
+    call = caller_env(2)
```

Comment:
Using `caller_env(2)` as `caller_env()` shows an unhelpful ".local()" frame:

```r
test_con("SQLITE", attributes = list(boop = "bop"))
#> Error in `.local()`:
#> ! `attributes` does not support the connection attribute "boop".
#> ℹ Allowed connection attribute is "azure_token".
```

with:

```
 3.   └─odbc::dbConnect(...)
 4.     └─odbc (local) .local(drv, ...)
 5.       └─odbc:::OdbcConnection(...) at odbc/R/dbi-driver.R:184:5
```

Reply:
FWIW, S4 adds the intermediate`.local` wrapper when the arguments of the generic and the method differ in some way. I think it happens here because our method adds the `dsn` argument before the docs. (No need to change anything, I just thought you might want to know where this was coming from)